### PR TITLE
[libc][math][c23] Add f16sqrt{,l,f128} C23 math functions

### PR DESCRIPTION
**Pull request by overmighty**
Part of #95250.

### CODE REVIEW

**Comment by overmighty** [comment]
cc @lntue

**Comment by overmighty** [comment]
Should probably change `fputil::sqrt` to handle NaNs correctly.

**Comment by overmighty** [comment]
Rebased onto `main` to catch errors like in #96976 and #97039.

File: libc/src/__support/FPUtil/generic/sqrt.h [hunk 1]
```diff
@@ -146,91 +144,19 @@ sqrt(InType x) {
...
-        // Round to nearest, ties to even
-        if (rb && (lsb || (r != 0)))
-          ++y_out;
-        break;
-      case FE_UPWARD:
-        if (rb || (r != 0) || sticky)
-          ++y_out;
-        break;
+        // Rounding bit.
+        y |= 1 << 1;
```
Comment (lntue) [inline libc/src/__support/FPUtil/generic/sqrt.h:153]: 2?

Comment (overmighty) [reply libc/src/__support/FPUtil/generic/sqrt.h:153]: Yes. See the code below:

```cpp
      // Sticky bit.
      y |= static_cast<unsigned int>(r != 0);
```

**Comment by overmighty** [comment]
Rebased to fix a merge conflict and added these 2 commits:
- [libc][math][c23] Move f16sqrt{,f,l} specs to llvm_libc_ext.td
- [libc][math][c23] Disable f16sqrt{l,f128} on AArch64 Linux

File: libc/src/__support/FPUtil/generic/sqrt.h (hunk 1 above)
Comment (lntue) [reply libc/src/__support/FPUtil/generic/sqrt.h:153]: Just put `y |= 2` is fine.
Also, can you add a TODO on the main for loop to reduce it to `OutFPBits::FRACTION_LEN + 2` or 3?

**Review by lntue — APPROVED** [review]

**Pull request merged**